js/huobipro.js

```diff
@@ -982,4 +987,124 @@ module.exports = class huobipro extends Exchange {
             }
         }
     }
+
+    async fetchDeposits (code = undefined, since = undefined, limit = undefined, params = {}) {
+        if (code === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchDeposits() requires a code argument');
+        }
+        if (limit === undefined || limit > 100) {
+            limit = 100;
+        }
+        await this.loadMarkets ();
+        let currency = undefined;
+        const request = {};
+        currency = this.currency (code);
+        request['currency'] = currency['id'];
+        request['type'] = 'deposit';
+        request['from'] = 0; // From 'id' ... if you want to get results after a particular transaction id, pass the id in params.from
+        request['size'] = limit; // Maximum transfers that can be fetched is 100
+        let response = await this.privateGetQueryDepositWithdraw (this.extend (request, params));
+        // return response
+        return this.parseTransactions (response['data'], currency, since, limit);
+    }
+
+    async fetchWithdrawals (code = undefined, since = undefined, limit = undefined, params = {}) {
+        if (code === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchWithdrawals() requires a code argument');
+        }
+        if (limit === undefined || limit > 100) {
+            limit = 100;
+        }
+        await this.loadMarkets ();
+        let currency = undefined;
+        const request = {};
+        currency = this.currency (code);
+        request['currency'] = currency['id'];
+        request['type'] = 'withdraw';
+        request['from'] = 0; // From 'id' ... if you want to get results after a particular Transaction id, pass the id in params.from
+        request['size'] = limit; // Maximum transfers that can be fetched is 100
+        let response = await this.privateGetQueryDepositWithdraw (this.extend (request, params));
+        // return response
+        return this.parseTransactions (response['data'], currency, since, limit);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        let timestamp = this.safeFloat (transaction, 'created-at');
+        if (timestamp !== undefined) {
+            timestamp = parseInt (timestamp);
+        }
+        let updated = this.safeFloat (transaction, 'updated-at');
+        if (updated !== undefined) {
+            updated = parseInt (updated);
+        }
+        let code = undefined;
```

Comment:
Please replace the code lines 1040 – 1051 with the following unified method call:
```let code = this.safeCurrencyCode (transaction, 'currency');```
See `safeCurrencyCode` implementation for details (it actually does the same).